api/src/main/java/org/openmrs/api/db/hibernate/HibernateProgramWorkflowDAO.java

```diff
@@ -136,11 +144,14 @@ public void deleteProgram(Program program) throws DAOException {
 	 */
 	@Override
 	public PatientProgram savePatientProgram(PatientProgram patientProgram) throws DAOException {
-		if (patientProgram.getPatientProgramId() == null) {
+		
+                if (patientProgram.getPatientProgramId() == null) {
 			sessionFactory.getCurrentSession().save(patientProgram);
 		} else {
 			sessionFactory.getCurrentSession().merge(patientProgram);
 		}
+                
+                CustomDatatypeUtil.saveAttributesIfNecessary(patientProgram);
```

Comment:
Shouldn't this be done before the patientProgram is saved, instead of after?

Reply:
Okay